catalog/views.py

```diff
@@ -1,3 +1,23 @@
 from django.shortcuts import render
+from .models import Book, Author, BookInstance, Genre
 
-# Create your views here.
+
+def index(request):
+    """View function for home page of site"""
+
+    # Generate counts of the main objects
+    num_books = Book.objects.count()
+    num_instances = BookInstance.objects.count()
+    num_instances_available = BookInstance.objects.filter(status__exact='a').count()# Available books (status = 'a')
```

Comment:
Tìm hiểu về các Field lookup khác ngoài `exact`

Reply:
một số loại `lookup` cơ bản:
`iexact`: Thực hiện tìm kiếm không phân biệt chữ hoa chữ thường.
`contains`: Tìm kiếm các mục chứa một chuỗi cụ thể.
`icontains`: Tìm kiếm các mục chứa một chuỗi cụ thể (không phân biệt chữ hoa chữ thường).
`in`: Tìm kiếm các mục có giá trị nằm trong danh sách các giá trị.
`gt`, `gte`, `lt`, `lte`: Thực hiện so sánh số học (lớn hơn, lớn hơn hoặc bằng, nhỏ hơn, nhỏ hơn hoặc bằng) với một giá trị cụ thể.
`range`: Tìm kiếm các mục có giá trị nằm trong một phạm vi giá trị.
`isnull`: Tìm kiếm các mục có giá trị của một trường là null hoặc không phải null.